libraries/AP_HAL_ChibiOS/system.cpp

```diff
@@ -304,6 +304,13 @@ void __entry_hook()
 }
 #endif
 
+uint32_t chibios_rand_generate()
+{
+    uint32_t val;
+    hal.util->get_random_vals((uint8_t*)&val, sizeof(val));
```

Comment:
that function can fail

Reply:
The `get_random_vals` has signature:     `virtual bool get_random_vals(uint8_t* data, size_t size) { return false; }`. This is a great example where `WARN_IF_UNUSED` would be helpful at compile time to warn to prevent misuse.

Reply:
hack pseudo code that isn't full-proof.. but is there such a full-proof solution:
```
WITH_SEMEPHORE()
uint32_t result;
for (uint8_t i=0; i< 10; i++) {
    if (get_random_hard(random_value)) {
        return random_value
    }
    delay_us(1) // or whatever time the RNG requires for various HW targets
}
get_random_soft(random_value)
return random_value
```